### ISSUE TITLE
SAVE "tpi:gethelp" crashes after issuing "N" to end process

### ISSUE THREAD

**Issue by rcalcagno** [timex-sinclair-projects/TS-Pico#7]
(from Kevin):
Whenever you use the gethelp command, it will leave the Pico in a non-responsive state.  While you get a "K" prompt, the Pico appears to
not have properly left the help routine.  A workaround for this is to do something with a LOAD command.  (It might work with any of the other SLVM commands too, haven't tested them yet.)  Even -> LOAD "" <- will be enough as the next command.  At this point, any tape command will generate a "J" error.  Then the "heartbeat" will resume.  The next command will function normally.

**Comment by rcalcagno** [comment]
The bug in this command is line #795 in tspico.py, part of the SEND_MSG2()function, in charge of sending back to the TS the response of a command  whose output is longer than one screen of info. I forgot that, upon pressing "N" key, the ROM immediately goes away, not taking any further info, so it doesn't "consume" the 0x00 I was sending back before ending; so that character got input during the next command cycle, corrupting it. The TS-Pico should recover from this next spurious command, and correctly processess the next one. Same happens with SAVE "tpi:getlog" as well, again, when the screen is longer than a single screen.
 
Interestingly, this same error happens when issuing LOAD "tpi:dir" with more than a screen full of data; however in that case the command ends correctly; that was the main reason I didn't catch this error before. The are only four functions that use SEND_MSG2(); all of them should work correctly now.
 
As a workaround, comment out the above mentioned line prepending a '#'. The final version should also get rid of lines # 1128, 1158 and 1162; VERBOSE was incorrectly disabled in GETHELP(), nevertheless these lines don't t affect the function behaviour.
 
Final note: incorrect emptying of TX/RX FIFOs on the Pico (which is this case) caused me many headaches; if you develop your own routines in uPython and suspect that's the case, this simple program on the TS side can help you determine the situation:
 
10 PRINT IN 14
20 OUT 14, 100
30 GOTO 10
 
This can help "unlock" things and bring the TS-Pico back to normal processing, by emtpying the TX FIFO and providing the RX FIFO with input. When you see the Pico flash the LED indicating a wrong command, hit CAPS+BREAK and then after a failed one, operation should be resumed.

**Comment by ryangray** [comment]
The problem of pressing anything other than 'n' at the end of the list causing it to not exit unless you press shift+space/break (and get a tape load error) still exists. Is this a different issue?

**Comment by rcalcagno** [comment]
I don't think this qualifies as a bug; it's more the impossibility to have a reliable way of getting out of the loop of answering "Y/N". If there's no more info to display, the user MUST press N, otherwise either the system times out or need to exit with Ctrl+Break. 

Even though I've tried a couple suggestions from Gus, none of them worked, so I wasn't able to come up with a way to avoiding this. Good thing is the system recovers in any case. I'm afraid that's where we are on this at this point

**Comment by ryangray** [comment]
I ran into this just now and had forgotten it was a known issue. If we still don't have a fix for this, I think we should make a note of it in the current documentation for gethelp along with the BREAK workaround. Also should list any other known bugs in the doc so users know what's already known.

**Comment by rcalcagno** [comment]
I will try this and confirm whether it's fixable or not.
